### PR TITLE
OutBuffer: remove unsafe write()

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: src/dmd/doc.d [hunk 1]
```diff
@@ -393,7 +393,7 @@ extern(C++) void gendocfile(Module m)
             // BUG: convert file contents to UTF-8 before use
             const data = buffer.data;
             //printf("file: '%.*s'\n", cast(int)data.length, data.ptr);
-            mbuf.write(data.ptr, data.length);
+            mbuf.write(data[0 .. data.length]);
```
Comment (Geod24) [inline src/dmd/doc.d]: `data` is already a slice

Comment (WalterBright) [reply src/dmd/doc.d]: ok

Comment (WalterBright) [reply src/dmd/doc.d]: I expect there are more of these, but this PR is about fixing the bottom. Fixing the next layer should be an additional PR.

I eventually want to make dmd @safe, and am starting with OutBuffer.

File: src/dmd/root/outbuffer.d [hunk 2]
```diff
@@ -83,11 +83,6 @@ struct OutBuffer
         notlinehead = true;
     }
 
-    extern (C++) void write(const(void)* data, size_t nbytes) pure nothrow
```
Comment (thewilsonator) [inline src/dmd/root/outbuffer.d]: Please update the header.

**Review by thewilsonator — APPROVED** [review]
Otherwise looks good.

**Pull request merged**